django/contrib/admin/static/admin/css/changelists.css

```
@@ -261,7 +261,7 @@
     background-color: var(--selected-row);
 }
 
-#changelist table tbody tr:has(input[type=checkbox]:checked) {
+#changelist table tbody tr .action-checkbox:has(input[type=checkbox]:checked) {
```

As far as I'm aware, we should use
```suggestion
#changelist table tbody tr:has(input.action-select[type=checkbox]:checked) {
```
Also the same selector must be updated for `forced-colors`:
https://github.com/django/django/blob/7cc138a58f73c17f07cfaf459ef8e7677ac41ac0/django/contrib/admin/static/admin/css/changelists.css#L272

---

Let me check and update this.

---

Ah I see now… you can get the same behaviour by deleting the CSS declaration above with the comment "Once the :has() is supported… the tr.selected can be removed".

The correct behaviour I observed was due to that ^ polyfill for FF selecting the rest of the tr.

---

@felixxm you're right, I tried and it was looking same as your screenshot,  So updated code that you suggested and remove table tag from CSS for unnecessarily nesting selectors